lib/box/index.js

```diff
@@ -216,20 +216,17 @@ function escapeBackslash(path) {
 }
 
 function getHash(path) {
-  return new Promise((resolve, reject) => {
-    const src = createReadStream(path);
-    const hasher = createSha1Hash();
-    let hash_value;
-
-    src.pipe(hasher)
-      .on('data', chunk => {
-        hash_value = chunk.toString('hex');
-      })
-      .on('end', () => {
-        resolve(hash_value);
-      })
-      .on('error', reject);
+  const src = createReadStream(path);
+  const hasher = createSha1Hash();
+
+  const finishedPromise = new Promise((resolve, reject) => {
+    src.once('error', reject);
+    src.once('end', resolve);
   });
+
+  src.on('data', chunk => { hasher.update(chunk); });
+
+  return finishedPromise.then(() => hasher.digest('hex'));
```

Comment:
Would you mind revert the changes here? It seems that the changes caused the performance regression. Let's see if any improvements could be made by reverting changes here